tools/pybind11Tools.cmake

```diff
@@ -37,25 +37,74 @@ function(select_cxx_standard)
 endfunction()
 
 # Internal: find the appropriate LTO flag for this compiler
-macro(_pybind11_find_lto_flag output_var prefer_thin_lto)
-  if(${prefer_thin_lto})
-    # Check for ThinLTO support (Clang)
-    check_cxx_compiler_flag("-flto=thin" HAS_THIN_LTO_FLAG)
-    set(${output_var} $<${HAS_THIN_LTO_FLAG}:-flto=thin>)
+function(_pybind11_find_lto_flag cxxflags_out linkerflags_out prefer_thin_lto)
+  # Release type (upper case for simpler comparisons)
+  string(TOUPPER "${CMAKE_BUILD_TYPE}" U_CMAKE_BUILD_TYPE)
+
+  set(try_cxx_flags "")
+  set(try_linker_flags "")
+
+  if(U_CMAKE_BUILD_TYPE MATCHES REL AND CMAKE_CXX_COMPILER_ID MATCHES "^(GNU|Clang)$")
```

Comment:
I wonder if that was why @wjakob needed the ".. and in yet another project" workaround.